src/Cryptol/Eval/Concrete/Value.hs

```diff
@@ -316,6 +317,15 @@ instance Backend Concrete where
   znToInt _ _m x = pure x
   znEq _ _m x y = pure $! x == y
 
+  -- NB: under the precondition that `m` is prime,
+  -- the only values for which no inverse exists are
+  -- congruent to 0 modulo m.
+  znRecip sym m x
+    | r == 0    = raiseError sym DivideByZero
```

Comment:
Does `Integer.recipModInteger` return 0 if `x` is 0, or did you mean to test `x == 0` here?

Reply:
`Integer.recipModInteger` returns 0 whenever the input does not have a multiplicative inverse, including for 0.